src/System.Security.Cryptography.OpenSsl/src/System/Security/Cryptography/ECDsaOpenSsl.ImportExport.cs

```diff
@@ -0,0 +1,278 @@
+            {
+                includePrivate = false;
+            }
+            else
+            {
+                includePrivate = true;
+            }
+
+            SafeEcKeyHandle key;
+            if (parameters.Curve is ECPrimeCurve)
+            {
+                ECPrimeCurve curve = (ECPrimeCurve)parameters.Curve;
+
+                if (!ValidateExplicitCurve(curve))
+                    throw new CryptographicException(SR.Argument_InvalidValue);// Cryptography_InvalidECPrimeCurveParameters);
+
+                key = ImportPrimeCurveParameters(parameters, curve, includePrivate);
+            }
+            else if (parameters.Curve is ECCharacteristic2Curve)
+            {
+                ECCharacteristic2Curve curve = (ECCharacteristic2Curve)parameters.Curve;
+                try {
+                    //TODO if (!ValidateExplicitCurve(curve, cbKey))
+                    //    throw new CryptographicException(SR.Argument_InvalidValue);// Cryptography_InvalidECPrimeCurveParameters);
+                    key = ImportCharacteristic2CurveParameters(parameters, curve, includePrivate);
+                }catch(Exception e)
+                {
+                    Interop.Crypto.OutputDebugString(e.ToString() + e.StackTrace);
+                    key = null;
+                }
+            }
+            else if (parameters.Curve is ECNamedCurve)
+            {
+                ECNamedCurve curve = (ECNamedCurve)parameters.Curve;
+
+                if (!ValidateNamedCurve(curve))
+                    throw new CryptographicException(SR.Argument_InvalidValue);// Cryptography_InvalidECNamedCurveParameters);
+
+                key = ImportNamedCurveParameters(parameters, curve, includePrivate);
+            }
+            else
+            {
+                throw new NotSupportedException();
+            }
+
+            SetKey(key);
+        }
+
+        /// <summary>
+        ///     Exports the key used by the ECC object into an ECParameters object.
+        /// </summary>
+        public override ECParameters ExportExplicitCurveParameters(bool includePrivateParameters)
+        {
+            // todo char2 curve
+            SafeEcKeyHandle currentKey = _key.Value;
+
+            ECParameters ecparams = ExportExplicitCurveParameters(currentKey, includePrivateParameters);
+            return ecparams;
+        }
+
+        /// <summary>
+        ///     Exports the key used by the ECC object into an ECParameters object.
+        /// </summary>
+        public override ECParameters ExportNamedCurveParameters(bool includePrivateParameters)
+        {
+            SafeEcKeyHandle currentKey = _key.Value;
+
+            ECParameters ecparams = ExportNamedCurveParameters(currentKey, includePrivateParameters);
+            string keyOidValueName = Interop.Crypto.EcKeyGetOidValueCurveName(currentKey);
+            string keyOidFriendlyName = Interop.Crypto.EcKeyGetOidFriendyCurveName(keyOidValueName);
```

Comment:
`new Oid("SHA1")`.  The Value property now is "1.3.14.3.2.26", and FriendlyName is "sha1" (lowercase for this one, apparently).

`new Oid("1.3.14.3.2.26")` produces the same result.

`new Oid("Ed25529")` makes both Value and FriendlyName be "Ed25529", since it wasn't a dotted decimal and it didn't map.

`new Oid("1.3.3.1.2")` has a FriendlyName of null, and a Value of "1.3.3.1.2".

The single-string constructor's behavior is
* Value is never null
* If the input is a dotted decimal, Value is the input, and FriendlyName is whatever that maps to (or null on no match)
* If FromFriendlyName would throw, Value=FriendlyName=input
* Otherwise it is the same as FromFriendlyName (which means that case normalization may occur).

So always using the one-string version should just do what we want.